web_poet/serialization/functions.py

```diff
@@ -60,3 +71,32 @@ def _deserialize__Url(cls: Type[_Url], data: SerializedLeafData) -> _Url:
 
 
 register_serialization(_serialize__Url, _deserialize__Url)
+
+
+def _serialize_HttpClient(o: HttpClient) -> SerializedLeafData:
+    serialized_data: SerializedLeafData = {}
+    for response_key, response in o.saved_responses.items():
+        serialized_response = serialize_leaf(response)
+        key_prefix = response_key + "-"
+        for k, v in serialized_response.items():
+            serialized_data[key_prefix + k] = v
+    return serialized_data
+
+
+def _deserialize_HttpClient(
+    cls: Type[HttpClient], data: SerializedLeafData
+) -> HttpClient:
+    serialized_responses: Dict[str, SerializedLeafData] = {}
+    for k, v in data.items():
+        response_key, subkey = k.rsplit("-", 1)
+        serialized_responses.setdefault(response_key, {})[subkey] = v
+
+    result = cls(return_only_saved_responses=True)
+    for response_key, serialized_response in serialized_responses.items():
+        result.saved_responses[response_key] = deserialize_leaf(
```

Comment:
It seems to make sense to add saved_responses parameter to HttpClient.__init__, with an initial list of saved responses.